### DISCUSSION TITLE
I have a special question, is it possible to configure chrome to open as a child process so that when the dotnet process stops unexpectedly, the chrome process can automatically close?

### DISCUSSION THREAD

**Discussion by SmallDrogan** [hardkoded/puppeteer-sharp#2344] [General]

**Comment by IliaPodolko** [comment]
Faced the same problem. If you kill your process using TaskManager or if you stop the VS debugger - chrome.exe still lives.
I have not found a way to tell chrome to kill itself after timeout.
JS window.close does not work due to security restrictions.

I think I will have to write ProcessWatchDog which will keep an eye on my main process and chrome to kill the chrome when the main dies.

**Comment by IliaPodolko** [comment]
I found a way how to trick the browser.
You have to open new tab using JS and then create interval to close itself (`"setTimeout(() => window.close(), 5000)`).
It will be you last tab and after that the browser will close itself.
Example:



        public async Task OpenPageAsync(string url, int width, int height)
        {
            var viewportOptions = new ViewPortOptions
            {
                Width = width,
                Height = height
            };

            await _page.EvaluateExpressionAsync($"window.open('{url}')");
            
            var targets = _browser.Targets();
            var openedTarget = targets.FirstOrDefault(p => p.Url == url);

            await _page.CloseAsync();
            _page = await openedTarget.PageAsync();

            foreach (var target in targets.Where(t => t != openedTarget))
            {
                var page = await target.PageAsync();
                if (page != null)
                {
                    await page.CloseAsync();
                }
            }

            await _page.SetViewportAsync(viewportOptions);
            await _page.GoToAsync(url);
        }

        public async Task<Stream> ScreenshotStreamAsync()
        {
            var result = await _page.ScreenshotStreamAsync();
            await _page.EvaluateExpressionAsync($"setTimeout(() => window.close(), 5000)");
            return result;
        }

**Reply by IliaPodolko** [reply]
My way works only for headless=false. It does not work for headless=true.
For some reason when I close all pages/tabs - browser still lives in headless mode!!! 

What the difference, why headless version behaves differently. I cannot find solution anywhere.
